### PR TITLE
chore: MSVC CI

### PR DESCRIPTION
**Pull request by nitz**
## Proposed changes 

- To `CMakeLists.txt`: For MSVC, adds `/MP` enables parallel compilation, and `/wd4996` disables the would-be-error (due to `/sdl`) warning about using methods marked deprecated. Brings behavior inline with mingw-w64 build.
- Adds job for building with msvc to CI.

----

I'm not in love with the way I'm getting the SDL2 paths to cmake, but it gets the job done.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -84,26 +103,45 @@ jobs:
...
+              if: startsWith(matrix.shell, 'cmd')
+              run: |
+                curl -L -o SDL2-devel-2.0.20-VC.zip https://github.com/libsdl-org/SDL/releases/download/release-2.0.20/SDL2-devel-2.0.20-VC.zip
+                unzip SDL2-devel-2.0.20-VC.zip
+                mv "${{ github.workspace }}/SDL2-2.0.20" "${{ github.workspace }}/sdl2"
+                mklink /J "${{ github.workspace }}\sdl2\SDL2" "${{ github.workspace }}\sdl2\include"
+                
+            #- name: Add msbuild to PATH
+            #  if: startsWith(matrix.shell, 'cmd')
+            #  uses: microsoft/setup-msbuild@v1.1
```
Comment (pktiuk) [inline .github/workflows/main.yml:122]: It looks like something you forgot to remove

**Review by pktiuk — COMMENTED** [review]
Compiles and works. :)  
I added one comment to the place I am not sure about

**Comment by nitz** [comment]
You were completely right! I found that at action at the same time I found the other that set up the whole toolchain and just forgot to pull it out :)

**Comment by pktiuk** [comment]
Thanks again for your contributions :)

**Pull request merged**